magento/magento2
# Prevent SOAP fault from using non built-in entities

### Description (*)

Instead of using `htmlentities()` to escape text used in an XML document we should use `htmlspecialchars()`
This will ensure that fault reasons containing diacritics will not result in invalid XML.

### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#38189

### Manual testing scenarios (*)

1. Create a store and set the locale to de_DE
2. Create some products in that store
3. Use a SOAP client to add another product to the DE store using a url_key that already exists
4. The SOAP client should now return a valid SOAP fault with the correct error message: URL-Key für den ausgewählten Store existiert bereits.

### Questions or comments

I have added a new function to [the existing unit test](https://github.com/magento/magento2/blob/f6d4542518bf01f7d3ceca330d7290de62afcb4e/app/code/Magento/Webapi/Test/Unit/Model/Soap/FaultTest.php) but I am not sure if that is sufficient. Any feedback will be welcome.

### Contribution checklist (*)
